swarm/storage/types.go

```diff
@@ -285,3 +311,36 @@ func (c ChunkData) Size() int64 {
 func (c ChunkData) Data() []byte {
 	return c[8:]
 }
+
+type ChunkValidator interface {
+	Validate(key Key, data []byte) bool
+}
+
+// Provides method for validation of content address in chunks
+// Holds the corresponding hasher to create the address
+type ContentAddressValidator struct {
+	hashMu sync.Mutex
+	Hasher SwarmHash
+}
+
+// Constructor
+func NewContentAddressValidator(hash SwarmHash) *ContentAddressValidator {
+	return &ContentAddressValidator{
+		Hasher: hash,
+	}
+}
+
+// Validate that the given key is a valid content address for the given data
+func (self *ContentAddressValidator) Validate(key Key, data []byte) bool {
+	self.hashMu.Lock()
+	defer self.hashMu.Unlock()
+	self.Hasher.Reset()
```

Comment:
this duplicates hasherstore code 
or would... but its not even the same
plus my 2c about concurrency